### DISCUSSION TITLE
About the `revalidate` attribute some questions

### DISCUSSION THREAD

**Discussion by luffyZh** [vercel/next.js#16894] [Help]
Hello, abount the `revalidate` attribute of `getStaticProps` there are some questions? Who can help me out?

1 - Whether the `revalidate` attribute has a default value? 

If I don't set it‘s value, will the server continue to generate static pages instead of just generating pages once in the revalidate seconds, increasing the pressure on the server.

2 - The page re-generation timing.

For example, when I set 10 seconds and the res data changed in 5s, the request within 10s was still using the cache page, and the new request after 10s updated the page data, then when did the page is re-generated, 10s later or 5 - 10s?

**Comment by alanqchen** [comment]
1. I don't believe there's a default value. If you want to use incremental static re-generation, you must specify a time.

2. I'm not sure if I understood the question exactly, but I believe the answer would be that the page is regenerated after 10s:
   - The first user loads the page and is given the cached page, the page is re-generated in the background
   - Once the new page has been generated, the revalidate timer starts and the new page will be served on the next request
   - Thus, at the very least, 10 seconds must pass between each request before the new data will be displayed. Also, note that this may increase depending on how long it takes the server to generate the page

If you have any experience with using the stale-while-revalidate HTTP header, it works similarly.

**Reply by luffyZh** [reply]
@alanqchen Thanks for your answer very much~
 I'm trying  to use incremental static re-generation && swr in client fetch data. So I may need to set an appropriate revalidate time， I hope this plan will bring performance improvement.

**Reply by alanqchen** [reply]
No problem and it should certainly bring performance similar to serving a static page (since that is what it's still essentially doing). And don't worry about decreasing the revalidate time to below 10s if you need to, even 2 or 3s the server will still be able to handle fine in my experience.

**Reply by luffyZh** [reply]
OK, your answer gives me confidence ~ thank you very much ~
